library/Faktory/Connection.hs

```diff
@@ -82,6 +83,44 @@ connect ConnectionInfo {..} = bracketOnError open connectionClose pure
       , connectionUseSocks = Nothing
       }
 
+  connectTo' :: ConnectionContext -> ConnectionParams -> IO Connection
+  connectTo' cg cParams =
+    bracketOnError
+      (resolve' (connectionHostname cParams) (connectionPort cParams))
+      (close . fst)
+      ( \(h, _) ->
+        connectFromSocket cg h cParams
+      )
+
+  -- This is copy and pasted from the connectTo implement. The only change is
+  -- adding a keep alive socket option.
+  -- see: https://hackage.haskell.org/package/connection-0.3.1/docs/src/Network.Connection.html#connectTo
+  resolve' :: String -> PortNumber -> IO (Socket, SockAddr)
+  resolve' host port = do
+      let hints = defaultHints { addrFlags = [AI_ADDRCONFIG], addrSocketType = Stream }
+      addrs <- getAddrInfo (Just hints) (Just host) (Just $ show port)
+      firstSuccessful $ map tryToConnect addrs
+    where
+      tryToConnect addr =
+          bracketOnError
+            (socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr))
+            close
+            (\sock -> do
+              setSocketOption sock KeepAlive 1
+              S.connect sock (addrAddress addr)
+              return (sock, addrAddress addr)
```

Comment:
I know this is a near direct copy of `resolve` from `connectTo`, however `SockAddr` is never actually used outside of this context. The only reason to keep this tuple would be to avoid any additional divergence in this copied function.

```suggestion
              return sock
```